src/toolbar/ToolbarKebab.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { DropdownButton, MenuItem, HorizontalNavMenuItem } from 'patternfly-react';
+import { ToolbarClick } from './ToolbarClick';
+import './styles.scss';
+
+/*
+ * .btn-group.open .dropdown-toggle {
+ *    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
+ * }
+ */
+const ToolbarSubmenu = props => (
+  <HorizontalNavMenuItem
+    id={props.id}
+    title={
+      <ToolbarClick
+        hidden={props.hidden}
+        text={props.text}
+        title={props.title}
+        icon={props.icon}
+      />
+    }
+    dropdown
+    submenu
+    onMouseEnter={() => props.setIsOpenId(props.id)}
+    onItemClick={() => props.setIsOpenId(props.id)}
+    onToggle={() => null}
+    open={props.openId === props.id}
+  >
+    { props.items.filter(i => !i.hidden).map(i => (
+      <MenuItem
+        key={i.id}
+        eventKey={i.id}
+        onClick={props.onClick && i.enabled ? (() => props.onClick(i)) : null}
+      >
+        <ToolbarClick {...i} />
+      </MenuItem>
+    )) }
+  </HorizontalNavMenuItem>
+);
+
+ToolbarSubmenu.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.any),
+  id: PropTypes.string.isRequired,
+  text: PropTypes.string,
+  title: PropTypes.string.isRequired,
+  icon: PropTypes.string,
+  hidden: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  openId: PropTypes.string,
+  setIsOpenId: PropTypes.func.isRequired,
+};
+
+const KebabListItem = (item, onClick, openId, setIsOpenId) => {
+  if (item.type === 'separator') {
+    return <MenuItem key={item.id} disabled={!item.enabled} eventKey={item.id} divider />;
+  }
+
+  if (item.type === 'buttonSelect') {
+    return <ToolbarSubmenu key={item.id} {...item} onClick={onClick} onSelect={console.log} openId={openId} setIsOpenId={setIsOpenId} />;
+  }
+
+  return (
+    <MenuItem disabled={!item.enabled} eventKey={item.id} >
+      <ToolbarClick key={item.id} {...item} />
+    </MenuItem>
+  );
+};
+
+export const ToolbarKebab = (props) => {
+  const [openId, setIsOpenId] = useState(undefined);
+
+  return (
+    <div className="kebab">
+      <DropdownButton id="menu_kebab" title={<span className="fa fa-ellipsis-v" />}>
```

Comment:
Please provide here `onClick={() => setIsOpenId(undefined)}` to reset the openId when user closes/opens the kebab